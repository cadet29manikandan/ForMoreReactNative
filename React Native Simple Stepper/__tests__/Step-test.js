import React from 'react';
import Step from '../src/Step';
import renderer from 'react-test-renderer';

describe('Step', () => {
  describe('renders', () => {
    test('snapshot', () => {
      const wrapper = renderer.create(<Step />).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
    test('defaultProps onPress', () => {
      Step.defaultProps.onPress();
      expect(Step.defaultProps.onPress).toBeDefined();
    });
    test('defaultProps renderImage', () => {
      expect(Step.defaultProps.renderImage).toBeUndefined();
    });
    test('renderImage to be defined', () => {
      const wrapper = renderer.create(<Step renderImage={() => {}} />);
      expect(wrapper).toBeDefined();
    });
    test('onPress to be defined', () => {
      const wrapper = renderer.create(<Step onPress={() => {}} />);
      expect(wrapper).toBeDefined();
    });
  });
});
