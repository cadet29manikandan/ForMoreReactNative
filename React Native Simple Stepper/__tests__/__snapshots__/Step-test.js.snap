// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step renders snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Image
    source={Object {}}
    style={
      Array [
        Object {
          "height": 36,
          "width": 36,
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;
